### ISSUE TITLE
feature: finish adoption of definition of SyncTarget and Location

### ISSUE THREAD

**Issue by MikeSpreitzer** [kubestellar/kubestellar#848]
### Feature Description

Most of this work was done in #814 but there are comments in that PR and follow-ups that still need some attention.

### Proposed Solution

Follow up on all the comments on #814 and follow-ons.

### Want to contribute?

- [ ] I would like to work on this issue.

**Comment by fileppb** [comment]
@MikeSpreitzer  Is it not possible to create synctargets.edge.kcp.io in a workspace without having created an APIBinding? And isn't possible to delete an APIBinding without deleting synctargets.edge.kcp.io? see https://github.com/kubestellar/kubestellar/pull/814#discussion_r1279866804

**Comment by MikeSpreitzer** [comment]
We do still have a few references to stuff in kcp-dev/kcp.

The following are in code.

@fileppb 
@yana1205 

```console
(venv) mspreitz@mjs12 kubestellar % find * \( -path docs/venv -prune \) -or \( -name "*go" -exec grep /workload \{\} \; -print -exec echo \; \)
	workloadv1alpha1 "github.com/kcp-dev/kcp/pkg/apis/workload/v1alpha1"
cmd/syncer/options/options.go

	workloadcliplugin "github.com/kcp-dev/kcp/pkg/cliplugins/workload/plugin"
test/e2e/framework/kubestellar_syncer.go
```

The following are in comments.

```console
(venv) mspreitz@mjs12 kubestellar % find * \( -path docs/venv -prune \) -or \( -name "*go" -exec grep /scheduling \{\} \; -print -exec echo \; \)
// A destination is a [kcp Location](https://github.com/kcp-dev/kcp/blob/v0.8.2/pkg/apis/scheduling/v1alpha1/types_location.go#L50)
pkg/apis/edge/v1alpha1/customize.go

// see https://github.com/kcp-dev/kcp/blob/v0.11.0/pkg/client/clientset/versioned/typed/scheduling/v1alpha1/placement.go#L48-L49 ;
// see https://github.com/kcp-dev/kcp/blob/v0.11.0/pkg/client/clientset/versioned/cluster/typed/scheduling/v1alpha1/placement.go#L47-L48 ;
pkg/mailboxwatch/listwatch.go

	//schedulingv1a1informers "github.com/kcp-dev/kcp/pkg/client/informers/externalversions/scheduling/v1alpha1"
pkg/placement/main.go
```

**Comment by fileppb** [comment]
Just saw this. Will work on it today.

**Comment by fileppb** [comment]
@yana1205 @MikeSpreitzer I noticed 3 things:
1)  I see that pkg/cliplugins/kubestellar/syncer-gen/edgesync.go uses
 kcpclient "github.com/kcp-dev/kcp/pkg/client/clientset/versioned"
shouldn't it use 
kcpclient "github.com/kubestellar/pkg/client/clientset/versioned"
2) pkg/test/e2e/framework/kubestellar-syncer.go uses 
workloadcliplugin "github.com/kcp-dev/kcp/pkg/cliplugins/workload/plugin"
shouldn't it use
workloadcliplugin "github.com/kubestellar/pkg/cliplugins/kubestellar/syncer-gen"
3) cmd/syncer/options/options.go uses 
fs.StringVar(&options.SyncTargetName, "sync-target-name", options.SyncTargetName,
	fmt.Sprintf("ID of the -to cluster. Resources with this ID set in the %q label will be synced.",   workloadv1alpha1.ClusterResourceStateLabelPrefix+"<ClusterID>"))
However kubestellar doesn't have any attribute named ClusterResourceStateLabelPrefix. That attribute only exists in KCP.
@yana1205 How would you like to proceed?

**Comment by yana1205** [comment]
@fileppb 
1.  The code doesn't work if we just replace it with kubestellar logicalcluster clientset. Syncer codes depend on kcp logical cluster (`core.kcp.io/v1alpha1/logicalclusters`) that's created when a workspace is created. The kcp logical cluster is different from kubestellar logical cluster both syntactically and semantically. We need to remove the dependency from codes relating with Syncer/Syncer manifest generation.
2. We can replace them as your suggestion. 
3. We can remove `workloadv1alpha1.ClusterResourceStateLabelPrefix`. When I created KS Syncer, I referred to the kcp syncer code. I forgot to modify the description of option `--sync-target-name`. For KS Syncer,  `--sync-target-name` is just a name of syncer. There is no more meaning than it.
